src/test/java/org/arquillian/container/chameleon/ManualSettingsXmlTestCase.java

```diff
@@ -0,0 +1,95 @@
+package org.arquillian.container.chameleon;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.List;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.JUnitCore;
+import org.junit.runner.RunWith;
+import org.junit.runner.notification.Failure;
+import org.junit.runners.model.InitializationError;
+
+/**
+ * @author <a href="mailto:mjobanek@redhat.com">Matous Jobanek</a>
+ */
+public class ManualSettingsXmlTestCase {
```

Comment:
I think in this case you can clearly see potential race condition if you run surefire with classes and methods, right? Not only the files can be alternated but also system properties if these test methods are run in the same JVM process. 

As for marking others marked thread-safe or not, I think it's not worth an effort - it's probably better to fail fast and spot the problem rather than speculating. This marker annotation let surefire run the test sequentially even when configured to run class and methods in parallel.